### PR TITLE
Feat/dlmm vault

### PR DESCRIPTION
**Pull request by 00xSam**

### CODE REVIEW

File: ts-client/src/dlmm/index.ts [hunk 1]
```diff
@@ -4303,6 +4279,39 @@ export class DLMM {
     }).add(syncWithMarketPriceTx);
   }
 
+  public async getMaxPrice(
+    minPrice: string,
+    binArrays?: BinArrayAccount[]
+  ): Promise<string> {
+    const _binArrays = binArrays || (await this.getBinArrays());
```
Comment (andrewsource147) [inline ts-client/src/dlmm/index.ts:4286]: Why use `_binArrays `? I guessed _variable is used for ignore variable?

Comment (andrewsource147) [reply ts-client/src/dlmm/index.ts:4286]: Do you fetch all binArrays? caused that may crash client, we should only read the binArray having liquidity and it's bin id is the biggest.
That can be archived through bitmap

File: ts-client/src/dlmm/index.ts [hunk 2]
```diff
@@ -4303,6 +4279,39 @@ export class DLMM {
     }).add(syncWithMarketPriceTx);
   }
 
+  public async getMaxPrice(
+    minPrice: string,
```
Comment (andrewsource147) [inline ts-client/src/dlmm/index.ts:4283]: why we need input minPrice here? the function getMaxPrice doesn't need input parameter as minPrice

File: ts-client/src/dlmm/index.ts (hunk 1 above)
Comment (00xSam) [reply ts-client/src/dlmm/index.ts:4286]: > Do you fetch all binArrays? caused that may crash client, we should only read the binArray having liquidity and it's bin id is the biggest. That can be archived through bitmap

wdym? fetching all binArrays just using 1 gPA?

File: ts-client/src/dlmm/index.ts (hunk 2 above)
Comment (00xSam) [reply ts-client/src/dlmm/index.ts:4283]: that's true, we dont need this

**Review by codewithgun — APPROVED** [review]

**Pull request merged**